buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/YamlRestCompatTestPlugin.java

```diff
@@ -73,84 +74,54 @@ public void apply(Project project) {
         ElasticsearchCluster testCluster = createTestCluster(project, yamlCompatTestSourceSet);
         testCluster.setTestDistribution(TestDistribution.DEFAULT);
 
-        // TODO: once https://github.com/elastic/elasticsearch/pull/62473 lands refactor this to reference the checkoutDir as an artifact
-        int priorMajorVersion = VersionProperties.getElasticsearchVersion().getMajor() - 1;
-        final Path checkoutDir = project.findProject(":distribution:bwc:minor")
-            .getBuildDir()
-            .toPath()
-            .resolve("bwc")
-            .resolve("checkout-" + priorMajorVersion + ".x");
-
         // copy compatible rest specs
-        Configuration compatSpec = project.getConfigurations().create("compatSpec");
-        Configuration xpackCompatSpec = project.getConfigurations().create("xpackCompatSpec");
-        Configuration additionalCompatSpec = project.getConfigurations().create("additionalCompatSpec");
+        Configuration bwcMinorConfig = project.getConfigurations().create("bwcMinor");
+        Dependency bwcMinor = project.getDependencies().project(Map.of("path", ":distribution:bwc:minor", "configuration", "checkout"));
+
         Provider<CopyRestApiTask> copyCompatYamlSpecTask = project.getTasks()
             .register("copyRestApiCompatSpecsTask", CopyRestApiTask.class, task -> {
+                project.getDependencies().add(bwcMinorConfig.getName(), bwcMinor);
```

Comment:
Why are we doing this in the task configuration? I think this should be up where we create the configuration and dependency.

Reply:
+1